akka-http-core/src/main/java/akka/http/javadsl/model/headers/ModeledCustomHeaderCompanion.java

```diff
@@ -0,0 +1,43 @@
+package akka.http.javadsl.model.headers;
+
+import akka.http.javadsl.model.HttpHeader;
+import akka.util.Helpers;
+
+import java.util.Optional;
+
+/**
+ * Companion class for the {@link ModeledCustomHeader} class. It offers methods to create {@link ModeledCustomHeader}
+ * from {@link String} or {@link HttpHeader}.
+ */
+public abstract class ModeledCustomHeaderCompanion<H extends ModeledCustomHeader> {
```

Comment:
Agree, Factory is a better name, I guess I was taken back by the current bad press of Factory...